plugin.video.101barz/default.py

```diff
@@ -0,0 +1,232 @@
+        date = item['publish_up']
+        thumbnail_image_url = item['card_image']
+        youtube_id = item['youtube_id']
+        youtube_url = 'plugin://plugin.video.youtube/play/?video_id=%s' % youtube_id
+
+        add_link(title, youtube_url, 'play_video', thumbnail_image_url, date + "\n" + desc, length)
+
+    # add next page url
+    url_next = url
+    OFFSET = "offset="
+    start_pos = convertToUnicodeString(url_next).find(OFFSET)
+    if start_pos >= 0:
+        end_pos = convertToUnicodeString(url_next).find("&", start_pos)
+        old_offset_value = convertToUnicodeString(url_next)[start_pos + len(OFFSET):end_pos]
+        old_offset_value = int(old_offset_value)
+        new_offset_value = old_offset_value + 12
+        old_offset_value = str(old_offset_value)
+        new_offset_value = str(new_offset_value)
+        url_next = convertToUnicodeString(url_next).replace(OFFSET + convertToUnicodeString(old_offset_value), OFFSET +
+                                                            convertToUnicodeString(new_offset_value))
+
+        add_dir(LANGUAGE(30009), url_next, 'list_videos', os.path.join(IMAGES_PATH, 'next-page.png'))
+
+    xbmcplugin.endOfDirectory(PLUGIN_HANDLE)
+
+
+def search(url):
+    # decode the url
+    url = urllib.parse.unquote_plus(url)
+
+    keyboard = xbmc.Keyboard('', LANGUAGE(30008))
+    keyboard.doModal()
+    if keyboard.isConfirmed() and keyboard.getText():
+        search_string = keyboard.getText().replace(" ", "+")
+        search_url = BASE_URL + convertToUnicodeString(url).replace("<searchstring>", search_string)
+        list_videos(search_url)
+
+
+def play_video(video_url):
+    list_item = xbmcgui.ListItem(path=video_url)
+    return xbmcplugin.setResolvedUrl(PLUGIN_HANDLE, True, list_item)
+
+
+def show_dialog(title, text):
+    dialog = xbmcgui.Dialog()
+    dialog.ok(title, text)
+
+
+def get_url(url):
+    if convertToUnicodeString(url).find(BASE_URL) >= 0:
+        complete_url = url
+    else:
+        complete_url = BASE_URL + url
+
+    log("complete_url", complete_url)
+
+    response = requests.get(complete_url, headers=HEADERS, cookies=COOKIES)
+    html_source = response.text
+    html_source = convertToUnicodeString(html_source)
+
+    #log("html_source", html_source)
+
+    return html_source
+
+
+def parameters_string_to_dict(parameters):
+    """ Convert parameters encoded in a URL to a dict. """
+    param_dict = {}
+    if parameters:
+        param_pairs = parameters[1:].split("&")
+        for paramsPair in param_pairs:
+            param_splits = paramsPair.split('=')
+            if (len(param_splits)) == 2:
+                param_dict[param_splits[0]] = param_splits[1]
+    return param_dict
+
+
+def add_link(title, url, mode, thumbnail_image_url, desc="", duration=""):
+    u = sys.argv[0] + "?url=" + urllib.parse.quote_plus(url) + "&mode=" + convertToUnicodeString(mode)
+    liz = xbmcgui.ListItem(label=title, thumbnailImage=thumbnail_image_url)
```

Comment:
thumbnailImage is also deprecated (not only iconImage)

Reply:
Fixed.